meshpy/FourC/beam_potential.py

```diff
@@ -0,0 +1,256 @@
+from ..header_functions import get_yes_no
+from ..boundary_condition import BoundaryCondition
+
+
+class BeamPotential:
+
+    def __init__(
+        self,
+        input_file,
+        *,
+        pot_law_prefactor=None,
+        pot_law_exponent=None,
+        pot_law_line_charge_density=None,
+        pot_law_line_charge_density_funcs=None,
+    ):
+        """Initialize object to enable beam potential interactions.
+
+        Args
+        ----
+        input_file:
+            Input file of current problem setup.
+        pot_law_prefactors: float, int, np.array, list
+            Prefactors of a potential law in form of a power law. Same number
+            of prefactors and exponents/line charge densities/functions must be
+            provided!
+        pot_law_exponent: float, int, np.array, list
+            Exponents of a potential law in form of a power law. Same number
+            of exponents and prefactors/line charge densities/functions must be
+            provided!
+        pot_law_line_charge_density: float, int, np.array, list
+            Line charge densities of a potential law in form of a power law.
+            Same number of line charge densities and prefactors/exponents/functions
+            must be provided!
+        pot_law_line_charge_density_funcs:
+            Functions for line charge densities of a potential law in form of a
+            power law. Same number of functions and prefactors/exponents/line
+            charge densities must be provided!
+        """
+
+        self.input_file = input_file
+
+        # if only one potential law prefactor/exponent is present, convert it
+        # into a list for simplified usage
+        if isinstance(pot_law_prefactor, (float, int)):
+            pot_law_prefactor = [pot_law_prefactor]
+        if isinstance(pot_law_exponent, (float, int)):
+            pot_law_exponent = [pot_law_exponent]
+        if isinstance(pot_law_line_charge_density, (float, int)):
+            pot_law_line_charge_density = [pot_law_line_charge_density]
+
+        # check if same number of prefactors and exponents are provided
+        if (
+            not len(pot_law_prefactor)
+            == len(pot_law_exponent)
+            == len(pot_law_line_charge_density)
+        ):
+            raise ValueError(
+                "Number of potential law prefactors do not match potential law exponents!"
+            )
+
+        self.pot_law_prefactor = pot_law_prefactor
+        self.pot_law_exponent = pot_law_exponent
+        self.pot_law_line_charge_density = pot_law_line_charge_density
+        self.pot_law_line_charge_density_funcs = pot_law_line_charge_density_funcs
+
+    def add_header(
+        self,
+        *,
+        potential_type="Volume",
+        cutoff_radius=None,
+        evaluation_strategy=None,
+        regularization_type=None,
+        regularization_separation=None,
+        integration_segments=1,
+        gauss_points=10,
+        potential_reduction_length=-1,
```

Comment:
Does `-1` make sense here, or ist ist just a default "empty" value? In case of the latter, I would make the default value `None`.

Reply:
`-1` disables the potential reduction length, therefore, I would keep it disabled as a default value.